frontend/www/js/omegaup/components/arena/ContestCard.vue

```diff
@@ -0,0 +1,198 @@
+        <b-col>
+          <b-card-text>
+            <font-awesome-icon icon="users" />
+            {{ contest.contestants }}
+          </b-card-text>
+        </b-col>
+        <b-col>
+          <b-button
+            v-if="contestTab === 0 && contest.participating"
+            ref="contestButtonEnter"
+            :href="getContestURL(contest.alias)"
+            variant="primary"
+          >
+            <font-awesome-icon icon="sign-in-alt" />
+            {{ T.contestButtonEnter }}
+          </b-button>
+          <b-button
+            v-else-if="
+              (contestTab === 0 || contestTab === 1) && !contest.participating
+            "
+            ref="contestButtonSingUp"
+            :href="getContestURL(contest.alias)"
+            variant="primary"
+          >
+            <font-awesome-icon icon="sign-in-alt" />
+            {{ T.contestButtonSingUp }}
+          </b-button>
+          <b-dropdown v-else-if="contestTab === 2" variant="primary">
+            <template #button-content>
+              <font-awesome-icon icon="sign-in-alt" />
+              {{ T.contestButtonEnter }}
+            </template>
+            <b-dropdown-item :href="getVirtualContestURL(contest.alias)">{{
+              T.contestVirtualMode
+            }}</b-dropdown-item>
+            <b-dropdown-item :href="getPracticeContestURL(contest.alias)">{{
+              T.contestPracticeMode
+            }}</b-dropdown-item>
+          </b-dropdown>
+        </b-col>
+      </b-row>
+    </b-container>
+  </b-card>
+</template>
+
+<script lang="ts">
+import { Vue, Prop, Component } from 'vue-property-decorator';
+import { types } from '../../api_types';
+import * as time from '../../time';
+import * as ui from '../../ui';
+import T from '../../lang';
+
+// Import Bootstrap an BootstrapVue CSS files (order is important)
+import 'bootstrap/dist/css/bootstrap.css';
+import 'bootstrap-vue/dist/bootstrap-vue.css';
+
+import { ButtonPlugin, DropdownPlugin, LayoutPlugin } from 'bootstrap-vue';
+Vue.use(ButtonPlugin);
+Vue.use(DropdownPlugin);
+Vue.use(LayoutPlugin);
+
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
+import { fas } from '@fortawesome/free-solid-svg-icons';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { ContestTab } from './ContestListv2.vue';
+library.add(fas);
+
+@Component({
+  components: {
+    FontAwesomeIcon,
+  },
+})
+export default class ContestCard extends Vue {
+  @Prop() contest!: types.ContestListItem;
+  @Prop() contestTab!: ContestTab;
+
+  T = T;
+  ui = ui;
+
+  get finishContestDate(): string {
+    return this.contest.finish_time.toLocaleDateString();
+  }
+
+  get startContestDate(): string {
+    return this.contest.start_time.toLocaleDateString();
+  }
+
+  get contestDuration(): string {
+    return time.formatContestDuration(
+      this.contest.start_time,
+      this.contest.finish_time,
+    );
+  }
+
+  getContestURL(alias: string): string {
+    return `/arena/${alias}/`;
```

Comment:
```suggestion
    return `/arena/${encodeURIComponent(alias)}/`;
```

oops, se me pasó esto! cuando estemos construyendo URLs, siempre hay que escapar las cadenas.